Sprint 7 HomeWork.ver1

DmitryBrosalin opened the pull request:
Добавил новый класс FileBackedTaskManager для работы с файлом, который хранит в себе информацию по задачам.

DmitryBrosalin committed:
Sprint 7 HomeWork.ver1
Добавил новый класс FileBackedTaskManager для работы с файлом, который хранит в себе информацию по задачам.

DmitryBrosalin committed:
Sprint 7 HomeWork.ver1
Добавил новый класс FileBackedTaskManager для работы с файлом, который хранит в себе информацию по задачам.

slavrikk commented on src/managers/FileBackedTaskManager.java:
```diff
@@ -0,0 +1,140 @@
+import java.nio.file.Paths;
+
+public class FileBackedTaskManager extends InMemoryTaskManager {
+    File backedListOfTasks;
```
Внутренние поля класса лучше обозначать как `private`

slavrikk commented on src/managers/FileBackedTaskManager.java:
```diff
@@ -0,0 +1,140 @@
+public class FileBackedTaskManager extends InMemoryTaskManager {
+    File backedListOfTasks;
+
+    public FileBackedTaskManager() throws IOException {
```
Путь файла или сам файл лучше передавать через конструктор

slavrikk commented on src/managers/FileBackedTaskManager.java:
```diff
@@ -0,0 +1,140 @@
+    File backedListOfTasks;
+
+    public FileBackedTaskManager() throws IOException {
+        super();
```
Можно не вызывать родительский конструктор

slavrikk commented on src/managers/FileBackedTaskManager.java:
```diff
@@ -0,0 +1,140 @@
+        loadFromFile(backedListOfTasks);
+    }
+
+    public void loadFromFile(File file) {
```
Этот метод лучше сделать приватным, так как он вызывается только внутри конструктора

slavrikk commented on src/managers/FileBackedTaskManager.java:
```diff
@@ -0,0 +1,140 @@
+                counter++;
+            }
+        } catch (IOException ex) {
+            System.out.println("Произошла ошибка во время чтения файла.");
```
По заданию следует выбрасывать кастомное исключение при ошибках

slavrikk commented on src/managers/FileBackedTaskManager.java:
```diff
@@ -0,0 +1,140 @@
+        }
+    }
+
+    public void getTaskFromString(String stringTask) {
```
Лучше private, так как это внутренняя реализация класса

slavrikk commented on src/managers/FileBackedTaskManager.java:
```diff
@@ -0,0 +1,140 @@
+                fileWriter.write(subtask.toString());
+            }
+        } catch (IOException e) {
+            System.out.println("Произошла ошибка во время записи файла.");
```
По заданию следует выбрасывать кастомное исключение при ошибках

slavrikk commented on src/managers/FileBackedTaskManager.java:
```diff
@@ -0,0 +1,140 @@
+        }
+    }
+
+    public void save() {
```
Лучше private, так как это внутренняя реализация класса

slavrikk reviewed (commented):
По заданию так же следует создать статический метод:

Кроме метода сохранения, создайте статический метод static FileBackedTaskManager loadFromFile(File file), который будет восстанавливать данные менеджера из файла при запуске программы.

DmitryBrosalin committed:
Sprint 7 HomeWork.ver2
Добавил кастомное исключение при ошибках.

DmitryBrosalin committed:
Sprint 7 HomeWork.ver2
Добавил кастомное исключение при ошибках.
Переделал работу метода loadFromFile - теперь он статический, а также возвращает экземпляр FileBackedTaskManager.
